Packages/com.unity.inputsystem/InputSystem/NativeInputRuntime.cs

```diff
@@ -227,11 +227,17 @@ public float pollingFrequency
         public double currentTimeOffsetToRealtimeSinceStartup => NativeInputSystem.currentTimeOffsetToRealtimeSinceStartup;
         public float unscaledGameTime => Time.unscaledTime;
 
-        public bool runInBackground => Application.runInBackground ||
-        // certain platforms ignore the runInBackground flag and always run. Make sure we're
-        // not running on one of those.
-        // TODO: Add more platforms here as they're discovered.
-        Application.platform == RuntimePlatform.PS5;
+        public bool runInBackground
+        {
+            get =>
+                Application.runInBackground ||
+                // certain platforms ignore the runInBackground flag and always run. Make sure we're
+                // not running on one of those and set the values when running on specific platforms.
+                m_platformSpecificRunInBackground;
+            set => m_platformSpecificRunInBackground = value;
```

Comment:
I'm wondering if we could do this in a way that only the platform specific input system packages could override the value and not general users. To prevent users interfering and causing issues. 

I'm also wondering if we can generalise it to a method for applying other settings that might be platform specific.

At the moment the platform specific input system packages seem to only use the public APIs from the input system such as InputSystem.RegisterLayout

Reply:
That said I see InputSystem.s_Manager is internal so this is probably limited to the input system package. I see that the ios package references InputSystem.s_Manager.m_Runtime. However iOS plugin is the main package

https://github.com/Unity-Technologies/InputSystem/blob/6da074044b05034ebebb39c820aecd10c557fa31/Packages/com.unity.inputsystem/InputSystem/Plugins/iOS/iOSSupport.cs

Can this definitely be used by the platform specific packages ?

Reply:
Oh I see now that SetPlatformRunInBackgroundSetting is the public API to access this field that both packages and users can set.

Reply:
> I'm wondering if we could do this in a way that only the platform specific input system packages could override the value and not general users. To prevent users interfering and causing issues.

After some investigation, there's no easy way of doing with this without adding some more processes and proxy names/packages. I'll elaborate on Slack.

> Can this definitely be used by the platform-specific packages?

This was validated that it worked on Xbox.